wait-for.c

```diff
@@ -354,9 +355,8 @@ int main(int argc, const char **argv) {
 	}
 
 exit:
-	if (extrav != NULL) {
-		free(extrav);
-	}
+	free(dirpath_buffer);
+	free(extrav);
```

Comment:
Why remove the null check?

Reply:
Not necessarily. It's defensive programming. While standards dictate its a nop, several runtimes miss it and would crash.

Reply:
Yes, but these systems are even not C89 compliant.

Reply:
Regardless, the semantic advantages are there and regardless of c89 compliance we should still be generally defensive.